ui/app/AppLayouts/WalletV2/components/AddAccountPopup.qml

```diff
@@ -0,0 +1,210 @@
+            anchors.topMargin: marginBetweenInputs
+        }
+
+        Row {
+            id: accountNameInputRow
+            anchors.left: parent.left
+            anchors.leftMargin: 8
+            anchors.right: parent.right
+            anchors.rightMargin: 10
+            anchors.top: separator.bottom
+            anchors.topMargin: marginBetweenInputs
+            height: 82
+            spacing: 10
+            Item {
+                implicitWidth: 434
+                height: parent.height
+                Input {
+                    id: accountNameInput
+                    anchors.fill: parent
+                    customHeight: 56
+                    placeholderText: qsTrId("enter-an-account-name...")
+                    label: qsTrId("account-name")
+                    validationError: popup.accountNameValidationError
+                }
+            }
+            Item {
+                //emoji placeholder
+                width: 80
+                height: parent.height
+                StyledText {
+                    id: inputLabel
+                    text: "Emoji"
+                    font.weight: Font.Medium
+                    font.pixelSize: 13
+                    color: Style.current.textColor
+                }
+                Rectangle {
+                    width: parent.width
+                    height: 56
+                    anchors.top: inputLabel.bottom
+                    anchors.topMargin: 7
+                    radius: 10
+                    color: "pink"
+                    opacity: 0.6
+                }
+            }
+        }
+
+        Input {
+            id: passwordInput
+            anchors.left: parent.left
+            anchors.leftMargin: 8
+            anchors.right: parent.right
+            anchors.rightMargin: 10
+            implicitHeight: 82
+            customHeight: 56
+            anchors.top: accountNameInputRow.bottom
+            anchors.topMargin: marginBetweenInputs
+            placeholderText: qsTr("Give the account a secure password")
+            label: qsTr("Password")
+            textField.echoMode: TextInput.Password
+            validationError: popup.passwordValidationError
+        }
+    }
+    footer: StatusButton {
+        anchors.top: parent.top
+        anchors.right: parent.right
+        text: loading ? qsTrId("loading") : qsTrId("add-account")
+        enabled: !loading && (passwordInput.text !== "") && (accountNameInput.text !== "")
+                 && (keyOrSeedPhraseInput.correctWordCount || (keyOrSeedPhraseInput.text !== ""))
+
+        MessageDialog {
+            id: accountError
+            title: qsTr("Adding the account failed")
+            icon: StandardIcon.Critical
+            standardButtons: StandardButton.Ok
+        }
+
+        onClicked : {
+            loading = true;
+            if (!validate() || (seedPhraseInserted && !validateSeed())) {
+                errorSound.play();
+                return loading = false;
```

Comment:
I stuck for a few seconds here. :)
Maybe not just I?
wondering why other `return`s return nothing?

Reply:
good point, just copied the code from AddAccountWithPrivateKey.qml/AddAccountWithSeed.qml., it was a combined `loading = false; return; `. I've updated the code a bit.